dlt/common/runtime/telemetry.py

```diff
@@ -112,6 +114,65 @@ def _track(success: bool) -> None:
                 _track(False)
                 raise
 
-        return _wrap  # type: ignore
+        return cast(TFun, _wrap)
+
+    return decorator
+
+
+def with_dataset_access_telemetry() -> Callable[[TFun], TFun]:
+    """Track dataset access but only once per pipeline instance"""
+
+    def decorator(f: TFun) -> TFun:
+        def _wrap(*f_args: Any, **f_kwargs: Any) -> Any:
+            pipeline = f_args[0] if f_args else None
+            schema: Union[Schema, str, None] = f_kwargs.get("schema") or (
+                f_args[1] if len(f_args) > 1 else None
+            )
+
+            def _track_dataset_access_once(success: bool) -> None:
+                if pipeline and not getattr(pipeline, "_dataset_access_tracked", False):
+                    with contextlib.suppress(Exception):
+                        schema_name = None
+                        if schema is None:
+                            schema_name = getattr(pipeline, "default_schema_name", None)
+                        elif isinstance(schema, str):
+                            schema_name = schema
+                        elif isinstance(schema, Schema):
+                            schema_name = schema.name
+
+                        props = {
+                            "success": success,
+                            "destination_name": (
+                                pipeline.destination.destination_name
+                                if pipeline.destination
+                                else None
+                            ),
+                            "destination_type": (
+                                pipeline.destination.destination_type
+                                if pipeline.destination
+                                else None
+                            ),
+                            "dataset_name_hash": (
+                                digest128(pipeline.dataset_name) if pipeline.dataset_name else None
+                            ),
+                            "default_schema_name_hash": digest128(schema_name),
+                        }
+                        # resolve runtime config and init telemetry
+                        if not _TELEMETRY_STARTED:
```

Comment:
you do not need to do it. this already happened just before Pipeline object is instantiated